pkgs/by-name/in/install-grub-ng/src/builder.rs

```diff
@@ -0,0 +1,153 @@
+mod appearance;
+mod entries;
+mod install;
+
+use std::{
+    collections::HashSet,
+    fmt::Write as _,
+    fs,
+    os::{linux::fs::MetadataExt, unix::fs::PermissionsExt},
+    path::{Path, PathBuf},
+};
+
+use eyre::Result;
+
+use crate::{
+    config::{Config, Password},
+    grub::Grub,
+};
+
+pub struct Builder<'conf> {
+    config: Config<'conf>,
+
+    grub_boot: Grub,
+    grub_store: Option<Grub>,
+
+    default_config: &'conf Path,
+    pub copied: HashSet<PathBuf>,
+
+    dry_run: bool,
```

Comment:
It seems error prone to have `!dry_run` every where. Could be nice to have a wrapper for each `fs` function we use that checks the flag automatically.
A TODO is fine by me if you like the idea:
```suggestion
    // TODO: create wrappers for fs functions we use that check `$DRY_RUN` automatically
    dry_run: bool,
```

Reply:
I've been thinking about it and I think fs wrappers are not really enough since it also runs a lot of extra logic (e.g. running os-prober) when not in dry run mode